eng/templates/stages/mirror.yml

```diff
@@ -162,17 +162,20 @@ stages:
         # We want to match on both.
         tagGrepQuery="refs/tags/$tag_name\(\^{}\)\?$"
 
-        echo "git ls-remote --tags destination | grep -q $tagGrepQuery"
-        if git ls-remote --tags destination | grep -q "$tagGrepQuery"; then
+        # `grep -q` exits as soon as a match is found; if there are multiple matches
+        # `git ls-remote` would continue to output even after grep exits, resulting in a
+        # broken pipe error (exit code 141). Saving the result in a var fixes the issue
+        tags=$(git ls-remote --tags destination)
+        if echo "$tags" | grep -q "$tagGrepQuery"; then
```

Comment:
FYI, I often go with `grep $ALL_THE_ARGUMENTS >/dev/null 2>&1` to keep things simple. That still makes grep process all the input, but discards all output, just like what `-q` does.